setup.py

```diff
@@ -115,7 +115,7 @@ def build_extensions(self):
                       '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(config.upper(), build_dir),
                       '-DPYTHON_EXECUTABLE:FILEPATH=' + sys.executable]
 
-        make_args = ['-j8'] if not os.environ.get('MAKEFLAGS') else []
+        make_args = ['-j' + str(min(8, len(os.sched_getaffinity(0))))] if not os.environ.get('MAKEFLAGS') else []
```

Comment:
Looks like `sched_getaffinity` is only available on some UNIX platforms. What is the return value of it when not available?

Reply:
I see. The method doesn't exist in  other platforms, per https://github.com/python/cpython/issues/81781. I'll fix that.